### ISSUE TITLE
wkg oci push fails if the image ref contains any uppercase letters

### ISSUE THREAD

**Issue by yoshuawuyts** [bytecodealliance/wasm-pkg-tools#134]
https://github.com/WebAssembly/wasi-io/pull/94 was fun to debug - I'm not sure if this is a bug in `wkg` or a limitation of the OCI reference format?

**Comment by brooksmtownsend** [comment]
I've run into this before 😕

**Comment by yoshuawuyts** [comment]
I believe #40 was due to the remote, but this error was local to `wkg`?

At a minimum, a better error here from `wkg` would be helpful.

**Comment by joonas** [comment]
This is actually a requirement of the OCI Distribution spec, see the [Pulling manifests](https://specs.opencontainers.org/distribution-spec/#pulling-manifests) section for full explanation. In short:
> \<name\> refers to the namespace of the repository. ... Throughout this document, \<name\> MUST match the following regular expression:
>
> `[a-z0-9]+([._-][a-z0-9]+)*(/[a-z0-9]+([._-][a-z0-9]+)*)*`

**Comment by thomastaylor312** [comment]
Yeah I think I agree with @joonas' comment on https://github.com/bytecodealliance/wkg-github-action/pull/5. We might be able to expose this as a flag, but not sure I'd just want to lowercase things automatically
